tests/networkpolicy_test.go

```diff
@@ -1,173 +1,264 @@
-				},
-			}
-			var err error
-			policy, err = virtClient.NetworkingV1().NetworkPolicies(vmia.Namespace).Create(networkpolicy)
-			Expect(err).ToNot(HaveOccurred())
 		})
 
-		AfterEach(func() {
-			waitForNetworkPolicyDeletion(policy)
-		})
+		Context("and vms limited by Default-deny networkpolicy", func() {
+			var policy *networkv1.NetworkPolicy
 
-		It("[test_id:1511] should be failed to reach vmia from vmib", func() {
-			By("Connect vmia from vmib")
-			assertPingFail(vmib, vmia)
-		})
+			BeforeEach(func() {
+				// deny-by-default networkpolicy will deny all the traffice to the vms in the namespace
+				By("Create deny-by-default networkpolicy")
+				policy = &networkv1.NetworkPolicy{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: serverVMI.Namespace,
+						Name:      "deny-by-default",
+					},
+					Spec: networkv1.NetworkPolicySpec{
+						PodSelector: metav1.LabelSelector{},
+						Ingress:     []networkv1.NetworkPolicyIngressRule{},
+					},
+				}
+				createNetworkPolicy(policy)
+			})
+
+			AfterEach(func() {
+				waitForNetworkPolicyDeletion(policy)
+			})
+
+			It("[test_id:1511] should be failed to reach serverVMI from clientVMI", func() {
+				By("Connect serverVMI from clientVMI")
+				assertPingFail(clientVMI, serverVMI)
+			})
+
+			It("[test_id:1512] should be failed to reach clientVMI from serverVMI", func() {
+				By("Connect clientVMI from serverVMI")
+				assertPingFail(serverVMI, clientVMI)
+			})
+			It("[test_id:369] should deny http traffic for ports 80/81 from clientVMI to serverVMI", func() {
+				PingHTTPEventually(clientVMI, serverVMI, 80).ShouldNot(Succeed(), "should deny HTTP traffic from clientVMI go serverVMI with port 80")
+				PingHTTPEventually(clientVMI, serverVMI, 81).ShouldNot(Succeed(), "should deny HTTP traffic from clientVMI go serverVMI with port 81")
+			})
 
-		It("[test_id:1512] should be failed to reach vmib from vmia", func() {
-			By("Connect vmib from vmia")
-			assertPingFail(vmia, vmib)
 		})
 
-	})
+		Context("and vms limited by allow same namespace networkpolicy", func() {
+			var policy *networkv1.NetworkPolicy
 
-	Context("vms limited by allow same namespace networkpolicy", func() {
-		var policy *v1network.NetworkPolicy
-
-		BeforeEach(func() {
-			// allow-same-namespave networkpolicy will only allow the traffice inside the namespace
-			By("Create allow-same-namespace networkpolicy")
-			networkpolicy := &v1network.NetworkPolicy{
-				ObjectMeta: v13.ObjectMeta{
-					Name: "allow-same-namespace",
-				},
-				Spec: v1network.NetworkPolicySpec{
-					PodSelector: v13.LabelSelector{},
-					Ingress: []v1network.NetworkPolicyIngressRule{
-						{
-							From: []v1network.NetworkPolicyPeer{
-								{
-									PodSelector: &v13.LabelSelector{},
+			BeforeEach(func() {
+				// allow-same-namespace networkpolicy will only allow the traffice inside the namespace
+				By("Create allow-same-namespace networkpolicy")
+				policy = &networkv1.NetworkPolicy{
```

Comment:
i think it was better before,
since waitForNetworkPolicyDeletion was checking if policy != nil
and was asserting the Delete suceess,

here the policy might not be nil but the policy might not be created yet (if some one would add a code in between),
the previous state was more robust to code changes imo.

Reply:
I am not doing the same thing ? creat a policy, and using the same function at AfterEach.

Reply:
policy is used here to two things, the manifest, and after the creation, before that there was a distinction, i think we better distinct. 
and then the delete logic would be more accurate imo

Reply:
If future changes break the chagnes and .Create is not executed waitForNetworkPolicyDeletion will fail since policy is not there, and that's a good thing.

Reply:
you wont see failures of waitForNetworkPolicyDeletion, even assert you wont see in AfterEach, (from what i understood Miguel shown).
and at the moment it breaks the logic, and imho this change is not connected to this PR and not necessary

Reply:
Returning policy at createPolicy function now.